### PR TITLE
[android] Added logging for Platform socket

### PR DESCRIPTION
**Pull request by alexzatsepin**

### CODE REVIEW

File: android/src/com/mapswithme/maps/location/PlatformSocket.java [hunk 1]
```diff
@@ -117,17 +150,17 @@ public void close()
...
 
     try
     {
       mSocket.close();
-      sLogger.d("Socket has been closed: ", this);
+      LOGGER.d("Socket has been closed: ", this + "\n");
     } catch (IOException e)
     {
-      sLogger.e("Failed to close socket: ", this, e);
+      LOGGER.e(e, "Failed to close socket: ", this);
```
Comment (alexzatsepin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: Added. I missed this place.

Comment (alexzatsepin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: You are right about separate line. But I added additional separate lines to make reading the logs more easily/conveniently - http://clip2net.com/s/3FrlZRW . What do you think about it?

Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: Up to you, but from my point of view grep over log file will do the job.

**Comment by ygorshenin** [comment]
LGTM

File: android/src/com/mapswithme/maps/location/PlatformSocket.java [hunk 2]
```diff
@@ -75,28 +103,33 @@ private static boolean isPortAllowed(int port)
...
-        sLogger.e("Failed to create the ssl socket, mHost = ", host, " mPort = ", port, e);
+        LOGGER.e(e, "Failed to create the ssl socket, mHost = " + host + " mPort = " + port);
       }
     } else
     {
       try
       {
-        return new Socket(host, port);
+        socket = new Socket(host, port);
+        LOGGER.d("Ssl socket is created and ssl handshake is passed successfully");
```
Comment (milchakov) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: некорректная строчка, нужно убрать ssl

Comment (alexzatsepin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: Fixed

File: android/src/com/mapswithme/maps/location/PlatformSocket.java [hunk 3]
```diff
@@ -75,28 +103,33 @@ private static boolean isPortAllowed(int port)
...
       } catch (IOException e)
       {
-        sLogger.e("Failed to create the socket, mHost = ", host, " mPort = ", port, e);
+        LOGGER.e(e, "Failed to create the socket, mHost = " + host + " mPort = " + port);
       }
     }
 
-    return null;
+    return socket;
```
Comment (alexzatsepin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: done

File: android/src/com/mapswithme/maps/location/PlatformSocket.java [hunk 4]
```diff
@@ -31,26 +35,50 @@
...
+  }
+
+  @NonNull
+  private static Logger createLogger()
+  {
+    if (BuildConfig.BUILD_TYPE.equals("beta"))
+    {
+      String externalDir = StorageUtils.getExternalFilesDir();
+      if (!TextUtils.isEmpty(externalDir))
+        return new FileLogger(externalDir + "/socket.log");
```
Comment (alexzatsepin) [inline android/src/com/mapswithme/maps/location/PlatformSocket.java]: done

**Comment by alexzatsepin** [comment]
@milchakov PTAL

**Comment by milchakov** [comment]
LGTM

**Pull request merged**